src/coreComponents/physicsSolvers/fluidFlow/SinglePhaseBase.cpp

```diff
@@ -188,6 +193,9 @@ void SinglePhaseBase::initializePostInitialConditionsPreSubGroups()
 
   resetViews( mesh );
 
+  // Compute hydrostatic equilibrium in the regions for which corresponding field specification tag has been specified
+  computeHydrostaticEquilibrium();
```

Comment:
I may be wrong but I have a feeling that by calling it at this point it won't work for `SurfaceElements` (at least not for `FaceElements`) because those subRegions are populated later on. Somehow we should have dedicated event for the initialization that occurs after the first generation is triggered.